### ISSUE TITLE
Python 3 support

### ISSUE THREAD

**Issue by odinho** [stx-labs/stacks.js#109]
Currently you get a terrible error message if you try to run on Python 3 (#104).  We can easily make a better one.  But the bestest fix would be to support Python 3 as well as 2.7.  This issue is for tracking that.

**Comment by deeplook** [comment]
If you mean this error below, I've just run into it. Unfortunately, the classifiers in setup say nothing about which version of Python one should expect blockstack to work with. :-(

```
$ blockstack
Traceback (most recent call last):
  File "/Users/dinu/miniconda3/envs/blockchain/bin/blockstack", line 34, in <module>
    from blockstack_client.cli import run_cli, exit_with_error
  File "/Users/dinu/miniconda3/envs/blockchain/lib/python3.5/site-packages/blockstack_client/__init__.py", line 24, in <module>
    import accounts
ImportError: No module named 'accounts'
```

**Comment by odinho** [comment]
I've opened a pull request for specifying it only works with Python 2 in the setup.py at #113.  Jude actually made the command itself give an error message when run with Python 3 now, so the traceback should be a lot more friendly in the next version already.

Though the metadata idea was not bad at all :)

**Comment by muneeb-ali** [comment]
Thanks @Velmont! 

To add some context, adding support for Python 3 in blockstack CLI/client is probably the right way to go. Blockstack server is more complex and needs more work to be compatible with Python 3, but given that you can simply talk to it over HTTP/RPC there is less need to make it Python 3 compliant.

**Comment by shea256** [comment]
I'm going to pop a 1 Bitcoin bounty on this (anyone is welcome to boost this amount). I think it'd be really cool for someone to take this up as a project.

**Comment by odinho** [comment]
lol, that makes me stop working on it, because clearly someone else will do it now :laughing: 

Had a few pull requests lined up.  It goes a bit deep down the rabbit hole of dependencies, but all the things I've found so far are really very simple to fix.  Even just setting `use_2to3=True` in setup.py is enough in many cases.  Though many of the fixes 2to3 does could just as well work in a code base supporting both Py2 and Py3.

So there's some start help if someone wants to beat me to it.  Since I'm only doing it randomly in evenings when I feel like it ;)

**Comment by muneeb-ali** [comment]
I think the bounty is not widely publicized yet i.e., not announced on Twitter / reddit etc. You might have a head start given you're active in the Slack!

**Comment by ye** [comment]
Other than the #104 error reported, Python 3.x also puked at places where `print` is called like a statement rather than a function (see [PEP-3105](https://www.python.org/dev/peps/pep-3105/).

``` bash
(blockstack)[ye@Yes-MacBook-Pro ~]$ blockstack lookup fredwilson.id
  File "/Users/ye/.envs/blockstack/bin/blockstack", line 71
    print json.dumps(result, indent=4, sort_keys=True)
             ^
SyntaxError: invalid syntax
```

**Comment by odinho** [comment]
I started work on this and had a lot of patches lined up.  Wanted to have all tests working before starting to post.  Then I quit Opera and forgot to copy the changes over when I handed in the company computer. :P  So in other words, I don't have any WIP on this anymore.  I'll probably restart at some point if noone beats me to it.

That being said, how would we primarily want to have the Python 3 support?  There's first of all levels, 1) just `cli` support, 2) full blockstack support.  I was working on _2_, I wanted everything to run on Python 3.

About diff _1_ and _2_, the cli drags in a lot of dependencies, so it can be a bit hard to differentiate.  Which was why I thought it safer to just go the route of full Py3 support.

Then there's how to do it, there's a) minimal changes and using `2to3` automatically in setup.py, there's b) using `six` to make code run in both, and c) something else.  I was doing _a_.  Though I in fact like _b_ better, but it's also much more invasive.

Anyone has some extra ideas to any of the above?
